### ISSUE TITLE
Automating Google Play Store Translation via Translatewiki

### ISSUE THREAD

**Issue by rgaudin** [kiwix/overview#80]
This ticket tracks the [Kiwix GSoC 2023 Project](https://www.kiwix.org/en/support-us/code/google-summer-of-code/) **Automating Google Play Store Translation via Translatewiki** until code contributions and/or specific tickets requires creating its own dedicated repository (or falls into [zim-android](https://github.com/kiwix/kiwix-android)).

**Candidates, contributors**, this ticket is the preferred location to discuss this project. Ask your questions here and mentors (@kelson42) shall respond.

**Mandatory reads**:
- https://github.com/kiwix/overview/blob/main/CONTRIBUTING.md
- https://github.com/kiwix/overview/blob/main/CODE_OF_CONDUCT.md

---
## Automating Google Play Store Translation via Translatewiki

**Objective**: we want to automate the translation process of our Google Play store entries via Translatewiki. This will help in making the apps available in multiple languages, making it accessible to a larger audience.

**Approach**: we suggest using Kotlin and Gradle to develop a tool that can handle the translations. Your tool will extract the app’s text content and send it to Translatewiki for translation. After the translation is completed, the tool will automatically update the Google Play store entry with the translated text.

**Deliverables**:

- A Kotlin-based tool that can extract text content from Google Play store entries.
- Integration with Translatewiki for translation.
- Automated updating of Google Play store entries with translated text.
- A comprehensive guide for app developers to use the tool for their Google Play store entries.

**Skills required**:

- Good understanding of Kotlin and Gradle;
- be somewhat familiar with the Google Play store and Translatewiki.

**Difficulty**: Hard, but once you’ve figured it out it should go quickly and 175 hours or so should suffice.

**Comment by shahnoor-khan** [comment]
Hey I'm Shahnoor, I have good knowledge of Kotlin and worked as an android developer.  I'm new to open source but I'm ready to learn. I want to know wheather the tool that need be developed to translate play store content using translate, need to be a mobile application?
and Can you please point me to some pointers or demo to better understand the problem? :)

**Comment by shahnoor-khan** [comment]
Hey I'm Shahnoor, I have good knowledge of Kotlin and worked as an android developer.  I'm new to open source but I'm ready to learn. I want to know wheather the tool that need be developed to translate play store content using translate, need to be a mobile application?
and Can you please point me to some pointers or demo to better understand the problem? :)

**Comment by shahnoor-khan** [comment]
Hey I'm Shahnoor, I have good knowledge of Kotlin and worked as an android developer.  I'm new to open source but I'm ready to learn. I want to know wheather the tool that need be developed to translate play store content using translate, need to be a mobile application?
and Can you please point me to some pointers or demo to better understand the problem? :)

**Comment by criticalAY** [comment]
@kelson42 Could you drop hints as to how you would like to achieve it as the tool is a vague term and it is not at all clear what and how it will be achieved.

**Comment by kelson42** [comment]
@criticalAY I believe there is enough information given, if you have a precise question, ready to answer it.

**Comment by criticalAY** [comment]
@kelson42 how do you want this tool to be made I mean tool is a vague term like are you planning this to be integrated with the Android studio project itself?

**Comment by kelson42** [comment]
@criticalAY If you have no clue how to start with, and therefore feel unable to make a proposal, then the task is probably not for you.

**Comment by criticalAY** [comment]
I understand that but I also have a project idea so where can we discuss that?

**Comment by Omkar76** [comment]
Good idea. Is the translation for playstore entries already available on translatewiki? I did some searching on kiwix desktop and android repos and found there is translation available for app content. https://translatewiki.net/wiki/Translating:Kiwix but didn't find anything for playstore entries. I can see translated entries on playstore [this german page for example](https://play.google.com/store/apps/details?id=org.kiwix.kiwixmobile&hl=de) where is this located on translatewiki?

I've got a rough idea of how it can be done. need to do some experimentation. Please share link of translatewiki page for playstore entries. Thanks.

**Comment by prasher1421** [comment]
I have a basic doubt whether you are going for KMM (Kotlin Multiplatform Mobile) or traditional Kotlin. Otherwise I have a precise plan of how should I start with the module.

**Comment by Keerti143** [comment]
I am Keerthana L, a computer science student, I have just entered my 6th semester at city engineering college Bengaluru. I am new to open source contributions but I am aware of python, C++, HTML,CSS and java script. I would love to contribute to your organization but would you please tell me how to get started?
Hoping to hear from you soon.
Regards
Keerthana L

**Comment by Rammuni28** [comment]
Hello sir,
I can help you get started with using Kotlin and Gradle to develop a translation tool that extracts text content from an app and sends it to Translatewiki for translation.it required Some necessary dependencies to add to our build.gradle file and some libraries also and a translation logic   .I have an idea about it how we can do it , sir kindly please give me any contact where i can tell you how we can do this i have proper road map for it.

**Comment by PK284** [comment]
Hello, @kelson42 I went through every detail mentioned in the Document.
I have 2 questions to ask you regarding this.
      1. Are there any best practices or guidelines that I should follow during the project?
      2. How will my work be evaluated and assessed during the project? How often am I supposed to communicate with my 
          mentors for feedback or help? 
          
I am drafting my proposal for the Gsoc:2023 and I am interested to work on this project.
Kindly help me so that I can submit my proposal ASAP.
Thanks
PK284

**Comment by Thre4dripper** [comment]
Hello @kelson42, As a full stack Android developer with over 6 years of experience, I'm excited to discuss my proposal for GSOC.
I'm confident in my abilities to complete the project  and create a comprehensive guide for app developers. I am founder developer in BeaBrand Company (A Next Level Ecommerce Startup). I know industry practices and doing android since 2018.

As a Smart India Hackathon Finalist ( The biggest International level Hackathon ) I know how to work in sprints, so you can rely on me.

I am interested to work on this project.
Thanks
